crates/apub/src/activities/deletion/delete.rs

```diff
@@ -38,38 +51,36 @@ use url::Url;
 /// wrapping it in an announce just like other activities, instead of having the community send it.
 #[derive(Clone, Debug, serde::Deserialize, serde::Serialize)]
 #[serde(rename_all = "camelCase")]
-pub struct DeletePostCommentOrCommunity {
-  to: PublicUrl,
+pub struct Delete {
+  pub(in crate::activities::deletion) to: PublicUrl,
   pub(in crate::activities::deletion) object: Url,
-  cc: [Url; 1],
+  pub(in crate::activities::deletion) cc: [Url; 1],
   #[serde(rename = "type")]
-  kind: DeleteType,
+  pub(in crate::activities::deletion) kind: DeleteType,
+  /// If summary is present, this is a mod action (Remove in Lemmy terms). Otherwise, its a user
+  /// deleting their own content.
+  pub(in crate::activities::deletion) summary: Option<String>,
```

Comment:
We distinguish between a delete and remove by whether a summary (remove reason) is present. Problem is that the remove reason is optional in Lemmy, so for now my workaround is to set it to an empty string if there is none. We definitely need a way to distinguish between them, and I cant think of any other.

Reply:
The type should be set as remove, no? Delete and Remove are two different apub terms.

https://www.w3.org/TR/activitystreams-vocabulary/#dfn-remove

Reply:
Its for removing an object from a collection or a group. I explained it in more detail in the code comment here: https://github.com/LemmyNet/lemmy/pull/1699#discussion_r688463022